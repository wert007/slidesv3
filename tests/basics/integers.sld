func printUintArray(array: uint[]) {
    print(array);
}

func main() {
    let a = 42;
    a = -32 + a;

    print(a);

    let b : int? = 32;
    if b == 32 {
        print('yay!');
    }

    let c : uint? = 32;
    if c == 32 {
        print('uyay!');
    }

    let d : uint = 3;
    let e : uint[] = [1, 2, d];
    printUintArray(e);

    let f = [1, 2, d];
    printUintArray(f);

    let g = cast a : uint;
    print(g);

    let h = cast -a : uint;
    print(h);
}