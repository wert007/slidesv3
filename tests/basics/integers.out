10
yay!
uyay!
[ 1, 2, 3 ]
[ 1, 2, 3 ]
10
none
0: 0
1: 0
2: 0
3: 0
4: 0
5: 0
6: 0
7: 0
= -1
